data-prepper-core/src/test/java/org/opensearch/dataprepper/parser/model/PipelineConfigurationTests.java

```diff
@@ -5,36 +5,53 @@
 
 package org.opensearch.dataprepper.parser.model;
 
+import com.amazon.dataprepper.model.configuration.PipelineModel;
+import com.amazon.dataprepper.model.configuration.PluginModel;
 import com.amazon.dataprepper.model.configuration.PluginSetting;
 import com.amazon.dataprepper.plugins.buffer.blockingbuffer.BlockingBuffer;
-import org.opensearch.dataprepper.TestDataProvider;
 import org.hamcrest.CoreMatchers;
-import org.junit.Test;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.opensearch.dataprepper.TestDataProvider;
 
 import java.util.ArrayList;
-import java.util.InputMismatchException;
+import java.util.Collections;
 import java.util.List;
 import java.util.Map;
-import java.util.Set;
 
 import static org.hamcrest.CoreMatchers.equalTo;
 import static org.hamcrest.CoreMatchers.is;
 import static org.hamcrest.CoreMatchers.isA;
 import static org.hamcrest.CoreMatchers.notNullValue;
 import static org.hamcrest.MatcherAssert.assertThat;
-import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertThrows;
-import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class PipelineConfigurationTests {
+
+    private PluginModel source;
+    private List<PluginModel> processors;
+    private List<PluginModel> sinks;
 
-public class PipelineConfigurationTests {
+    @BeforeEach
+    void setUp() {
+        source = TestDataProvider.validSingleConfiguration();
+        processors = TestDataProvider.validMultipleConfigurationOfSizeOne();
+        sinks = TestDataProvider.validMultipleConfiguration();
+    }
 
     @Test
-    public void testPipelineConfigurationCreation() {
-        final PipelineConfiguration pipelineConfiguration = new PipelineConfiguration(TestDataProvider.validSingleConfiguration(),
-                null,
-                TestDataProvider.validMultipleConfigurationOfSizeOne(),
-                TestDataProvider.validMultipleConfiguration(),
-                TestDataProvider.TEST_WORKERS, TestDataProvider.TEST_DELAY);
+    void testPipelineConfigurationCreation() {
+        final PipelineModel pipelineModel = mock(PipelineModel.class);
+        when(pipelineModel.getSource()).thenReturn(source);
+        when(pipelineModel.getProcessors()).thenReturn(processors);
+        when(pipelineModel.getPreppers()).thenReturn(null);
```

Comment:
Do we still `preppers` for this when statement?

Reply:
I might be able to remove this line. I originally had to set this to `null` for preppers to work. But, then some other issues came up, so I removed any support for prepper: in the PipelineConfiguration class.